frontend/src/components/FragmentComment.js

```diff
@@ -0,0 +1,50 @@
+import '../styles/FragmentComment.css';
+import React from 'react';
+
+function FragmentComment({ children }) {
+  try {
+    const citationRegex = /^\[.*?\]\s*\n(.*)$/m;
+    if (citationRegex.test(children)) {
+      let [citation, commentaire] = children[0].split(/\n/);
+      citation = citation.replace(/[\[\]]/g, '');
+      return <p
+        onMouseEnter={
+          e => {
+            e.preventDefault();
+            e.stopPropagation();
+            highlight(citation);
+          }
+        }
+        onMouseLeave={
+          e => {
+            e.preventDefault();
+            e.stopPropagation();
+            unHighlight();
+          }
+        }
+        className="fragmentComment"
+        title="Highlight in document"
+      >{commentaire}</p>;
+    }
+  } catch (e) {}
+
+  function highlight(texte) {
+    let passages = document.getElementsByClassName('main col');
+    for (let passage of passages) {
+      if (passage.innerHTML.includes(texte)) {
+        passage.innerHTML = passage.innerHTML.replace(texte, '<highlighted>' + texte + '</highlighted>');
+      }
+    }
+  }
+
+  function unHighlight() {
+    let passages = document.getElementsByClassName('main col');
+    for (let passage of passages) {
+      passage.innerHTML = passage.innerHTML.replace('<highlighted>', '');
+      passage.innerHTML = passage.innerHTML.replace('</highlighted>', '');
```

Comment:
How comes that this piece of code is duplicated? Are both instances really necessary?

Reply:
This piece of code is now cleaned of any duplication in the last commit.